### DISCUSSION TITLE
use style-loader from webpack 4

### DISCUSSION THREAD

**Discussion by Mohammadgeek** [webpack/webpack#17197] [General]
Hello, I must use webpack4 for my project but problem to use style-loader & css-loader 
please reply to this question, thanks a lot!

## JSON template 
```
{
  "name": "myportofilo",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "webpack": "webpack",
    "runsass": "sass ./scss/style.scss ./css/style.css --watch"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@webpack-cli/serve": "^2.0.4",
    "css-loader": "^6.7.3",
    "file-loader": "^6.2.0",
    "html-loader": "^0.5.5",
    "html-webpack-plugin": "^4.5.2",
    "node-sass": "^8.0.0",
    "sass-loader": "^13.2.0",
    "style-loader": "^3.3.2",
    "webpack": "^4.25.0",
    "webpack-cli": "^3.3.4",
    "webpack-dev-server": "^4.15.0"
  }
}
```
## error in web pack cli 
```
ERROR in ./js/style.css
Module build failed (from ./node_modules/style-loader/dist/cjs.js):
TypeError: this.getOptions is not a function
    at Object. loader

```

how to fix these problems? if upgrade the web pack to version 5 solve these problems or not?

**Comment by alexander-akait** [comment]
You need to use `style-loader@2` version for webpack v4

**Reply by Mohammadgeek** [reply]
> You need to use `style-loader@2` version for webpack v4

thanks for the reply I uninstall the style-loader and install version 2 but problem for css-loader version!

which css-loader & MiniCssExtractPlugin are compatible with version 4?

**Reply by alexander-akait** [reply]
v5 https://github.com/webpack-contrib/css-loader/blob/v5.2.7/package.json#L42
